EventFilter/HcalRawToDigi/src/HcalUnpacker.cc

```diff
@@ -695,18 +712,33 @@ void HcalUnpacker::unpackUTCA(const FEDRawData& raw, const HcalElectronicsMap& e
 	  colls.qie10 = new QIE10DigiCollection(ns);
 	}
 	else if (colls.qie10->samples() != ns) {
-	  // This is horrible
-	  edm::LogError("Invalid Data") << "QIE10 Collection has " << colls.qie10->samples() << " samples per digi, raw data has " << ns << "!";
-	  return;
+          // if this sample type hasn't been requested to be saved
+          // warn the user to provide a configuration that prompts it to be saved
+          if( colls.qie10Addtl.find( ns ) == colls.qie10Addtl.end() ) {
+	    edm::LogError("Invalid Data") << "The default QIE10 Collection has " 
```

Comment:
since this message is the same as the QIE11 one, maybe it could be made into a function with arguments for the collection name and # samples?